packages/discord.js/src/structures/ThreadChannel.js

```diff
@@ -422,6 +422,22 @@ class ThreadChannel extends BaseChannel {
     return this.edit({ locked, reason });
   }
 
+  /**
+   * Sets archived and whether the thread can be **unarchived** by anyone with the
+   * {@link PermissionFlagsBits.SendMessages} permission. When a thread is locked, only members with the
+   * {@link PermissionFlagsBits.ManageThreads} permission can unarchive it.
+   * @param {string} reason Reason for locking or unlocking the thread
+   * @returns {Promise<ThreadChannel>}
+   * @example
+   * // Set the thread to locked and archived
+   * thread.closeAndLock()
+   *   .then(() => console.log(`Thread is now locked and archived.`))
+   *   .catch(console.error);
+   */
+  closeAndLock(reason) {
```

Comment:
not a fan of the name.

i think a better soloution would be to add a `lock` parameter to setArchive?

Reply:
I agree setArchived should have three potential signatures,
setArchived() => archives
setArchived(false, lock: never) => unarchived
setArchived(true, lock: boolean) => archive and set lock accordingly